token.go

```diff
@@ -82,9 +84,11 @@ func (t *Token) SigningString() (string, error) {
 	return strings.Join(parts, "."), nil
 }
 
-// Parse parses, validates, and returns a token.
-// keyFunc will receive the parsed token and should return the key for validating.
-// If everything is kosher, err will be nil
+// Parse parses, validates, verifies the signature and returns the parsed token.
+// keyFunc will receive the parsed token and should return the cryptographic key
+// for verifying the signature.
+// keyFunc should validate the 'alg' claim in the token matches the expected algorithm.
+// If everything is kosher, err will be nil.
```

Comment:
I would link the following for more information on the main reason, as it contains the vulnerability it may cause on unprotected code https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/

Also a nitpick: I'd change _kosher_ for OK; while we both know the meaning of the word not all audiences may be familiar with it and we want the docs to be as neutral as possible on terminologies

Reply:
Agree about the word kosher, I will change it.

Reply:
How about just removing the sentence "If everything is kosher, err will be nil.". Looking at the golang source code, functions that may return errors do not explicitly state something like "return error if xyz problem occurred". It would be fairly obvious and repetitive.